kubespawner/spawner.py

```diff
@@ -116,21 +117,26 @@ class KubeSpawner(Spawner):
     # Reflectors keeping track of the k8s api-server's state for various k8s
     # resources are singletons as that state can be tracked and shared by all
     # KubeSpawner objects.
-    reflectors = {
-        "pods": None,
-        "events": None,
-    }
+    reflectors = {}
 
     # Characters as defined by safe for DNS
     # Note: '-' is not in safe_chars, as it is being used as escape character
     safe_chars = set(string.ascii_lowercase + string.digits)
 
+    def _get_reflector_key(self, kind: str) -> Tuple[str, str, Optional[str]]:
+        if self.enable_user_namespaces:
+            # one reflector fo all namespaces
+            return (kind, None)
+
+        return (kind, self.namespace)
+
     @property
     def pod_reflector(self):
         """
         A convenience alias to the class variable reflectors['pods'].
```

Comment:
pod_reflector and event_reflector includes outdated docstrings

Reply:
Fixed